rails-commenting.rb

```diff
@@ -7,56 +7,100 @@
 # FILE: app/controller/blog_posts_controller.rb
 
 # ---1)
+# Taking place here in this code is 
+
+# BlogPostController is the controller. The controller is a logical center of an application. It will organize the communication between the user, views, and the model. A controller is made using $rails g controller (Controllername). In the next line, there is a def method with the name index. Index means to send back the entire list of data available for the model being queried. So the method's name is index, and index has a defintion. 
+
+
+
 class BlogPostsController < ApplicationController
   def index
+
+
     # ---2)
+
+# There is an instance variable named post is making an Active Record call on all of the BlogPost content and saving it in the post instance variable. 
+
     @posts = BlogPost.all
   end
 
   # ---3)
+
+# This is a method called show, that will display one item, by accessing it with a param id, after the variable instance of post. This will allow us to access an id inside of the item, and display/show it. 
   def show
     @post = BlogPost.find(params[:id])
   end
 
   # ---4)
+
+  # This is a method called new. New will allow users to add information into our web site, and store the information into the database. Create will allow the user to take data and add to it, to be in the database, creating a new instance. 
+
   def new
     @post = BlogPost.new
   end
 
   def create
+
+
     # ---5)
+
+# The instance of post which is BlogPost.all is allowing the users to create an instance whenever the user making a new entry under the blog_post_params in the Active Record. If the post meets all of the parameters, (being a strong param, meaning this will dictate the attributes allowed, so if all of the necessary items are met and deemed valid, than this will happen, which is essentially a conditional statement),than it will redirect to the _path. The path can link to another website, home screen, or confirmation screen,, etc. 
+
+# Than there is a def edit. Edit will be sends overs HTML forms, which will allow user to fill it out and send ouuut overwrite into a database. 
+
     @post = BlogPost.create(blog_post_params)
     if @post.valid?
       redirect_to blog_post_path(@post)
     end
   end
 
   def edit
+   
     # ---6)
+ 
+# Here we are again using the post instance, using the .find after BlogPost to find one item by its id. Since this is under the def edit, it is used to edit an existing model object. 
```

Comment:
One important note on edit and new is that these two methods will essentially call on a form.